lib/message_bus.rb

```diff
@@ -248,20 +248,21 @@ def publish(channel, data, opts = nil)
 
     channel_opts = nil
 
-    if opts && ((age = opts[:max_backlog_age]) || (size = opts[:max_backlog_size]))
+    if opts && (opts[:max_backlog_age] || opts[:max_backlog_size])
       channel_opts = {
-        max_backlog_size: size,
-        max_backlog_age: age
+        max_backlog_age: opts[:max_backlog_age],
+        max_backlog_size: opts[:max_backlog_size]
       }
     end
 
     encoded_channel_name = encode_channel_name(channel, site_id)
     reliable_pub_sub.publish(encoded_channel_name, encoded_data, channel_opts)
   end
 
-  def blocking_subscribe(channel = nil, &blk)
+  def blocking_subscribe(channel = nil, site_id = nil, &blk)
```

Comment:
@pawelniewie Why this method in particular? There are several other methods which could have this change made, and indeed be tested as such. Would you mind splitting this out into a separate PR?

Reply:
also api wise I generally prefer `blocking_subscribe(channel = nil, site_id: nil, &blk)` these days but that is probably part of a larger discussion.

Reply:
@pawelniewie When you split that out, please also rebase on master and add appropriate documentation.

Reply:
@benlangfeld I think we should first try to get diagnostics / redis stream out, not against an overhaul but lets close off those things first. also probably worth tracking how much we would break on such an overhaul separately.  its a bit tricky as a pattern cause usually if there is only 1 param I really like unamed, and in some cases even N params can be perfect unamed ... eg: max_of(1,2,3,4)